### PR TITLE
Handle reset identities event

### PR DESCRIPTION
**Pull request by addb**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by emdobrin — COMMENTED** [review]
@addb another use-case we should handle is the async processing of the server response: 
1. a network request was initiated 
2. a reset event was received and processed - resetting identities
3. the response for the network request (1) was received and processResponseData was executed after reset
Step 3 should not set any identities that were previously reset in step 2.

**Comment by addb** [comment]
> @addb another use-case we should handle is the async processing of the server response:
> 
> 1. a network request was initiated
> 2. a reset event was received and processed - resetting identities
> 3. the response for the network request (1) was received and processResponseData was executed after reset
>    Step 3 should not set any identities that were previously reset in step 2.

@emdobrin I wanted to get some feedback on changes I made in Analytics [Drop hit being processed if reset is called](https://github.com/adobe/aepsdk-analytics-ios/pull/41/commits/4119f55a61852f62d419951d0680938b2c9bb503#diff-f8652d7bd13060084dd0358e5d46bc47665c65118cdd589e3e653fd4524abdb7R60) and [Dispatch response only for events triggered after reset](https://github.com/adobe/aepsdk-analytics-ios/pull/41/commits/4119f55a61852f62d419951d0680938b2c9bb503#diff-f8652d7bd13060084dd0358e5d46bc47665c65118cdd589e3e653fd4524abdb7R151)

File: AEPAudience/Sources/Audience.swift [hunk 1]
```diff
@@ -130,7 +132,9 @@ public class Audience: NSObject, Extension {
     /// - Parameter event: The event coming from the reset API invocation
     private func handleAudienceResetRequest(event: Event) {
         Log.debug(label: getLogTagWith(functionName: #function), "Received an Audience Manager reset event, clearing all stored Audience Manager identities and visitor profile.")
-        state?.clearIdentifiers()
+        state?.hitQueue.clear()
+        state?.clearAudienceIdentifiers()
+        state?.lastResetIdentitiesTimestamp = event.timestamp.timeIntervalSince1970
```
Comment (praveek) [inline AEPAudience/Sources/Audience.swift:137]: This should be in the other function.

File: AEPAudience/Sources/Audience.swift [hunk 2]
```diff
@@ -130,7 +132,8 @@ public class Audience: NSObject, Extension {
     /// - Parameter event: The event coming from the reset API invocation
     private func handleAudienceResetRequest(event: Event) {
         Log.debug(label: getLogTagWith(functionName: #function), "Received an Audience Manager reset event, clearing all stored Audience Manager identities and visitor profile.")
-        state?.clearIdentifiers()
+        state?.clearAudienceIdentifiers()
```
Comment (praveek) [inline AEPAudience/Sources/Audience.swift:135]: Can we move the logic to set the lastResetTimestamp internal to AudienceState

**Review by praveek — APPROVED** [review]

**Pull request merged**